base/inc/AdePT/SparseVector.h

```diff
@@ -718,12 +718,14 @@ class SparseVector : public SparseVectorInterface<Type> {
   }
 
 public:
+  template <copcore::BackendType backend>
   __host__ static SparseVector *MakeInstanceAt(void *addr = nullptr)
   {
     if (!addr) return new SparseVector(N);
-    bool devicePtr = copcore::is_device_pointer(addr);
-    if (devicePtr) return SparseVector::ConstructOnDevice(addr);
-    return new (addr) SparseVector(N);
+    if (backend == copcore::BackendType::CUDA)
+      return SparseVector::ConstructOnDevice(addr);
+    else
+      return new (addr) SparseVector(N);
   }
```

Comment:
This looks like a different change, can you submit this as a separate PR? Also, can we get rid of the broken `is_device_pointer` completely?

Reply:
Yes, I forgot to push this separately, the idea is to get rid of this function.

Reply:
Moved to #138